src/tiled/tileseteditor.cpp

```diff
@@ -435,8 +436,41 @@ void TilesetEditor::addTiles()
         QPixmap image;
     };
     QVector<LoadedFile> loadedFiles;
-
+    // If the tile is already in the tileset, warn user and confirm addition
+    bool dontAskAgain = false, rememberOption = true;
     for (const QString &file : files) {
+        bool addImage = true;
+        const QMap<int, Tile*> addedTiles = mCurrentTilesetDocument->tileset().data()->tiles();
+        for (auto tile : addedTiles) {
+            if (tile->imageSource() == file) {
+                if (dontAskAgain) {
+                    addImage = rememberOption;
+                    break;
+                }
+                QCheckBox *checkBox = new QCheckBox(tr("Apply this action to all tiles"));
+                QMessageBox warning(QMessageBox::Warning,
+                            tr("Add Tiles"),
+                            tr("Tile \"%1\" already exists in the tileset!").arg(file),
+                            QMessageBox::Yes | QMessageBox::No,
+                            mMainWindow->window());
+                warning.setDefaultButton(QMessageBox::Yes);
+                warning.setInformativeText(tr("Add anyway?"));
+                warning.setCheckBox(checkBox);
+                if (warning.exec() != QMessageBox::Yes) {
+                    rememberOption = addImage = false;
+                }
+                else {
+                    rememberOption = true;
+                }
+                if (checkBox->checkState() == Qt::Checked) {
+                    dontAskAgain = true;
+                }
+                break;
```

Comment:
If I only move that part I would still need the `addImage` variable, and if I am not in a loop I can't break on the `dontAskAgain` if, so I would need an if-else, is that ok? I can commit so you can take a look and decide what is best.